src/platform_impl/windows/mod.rs

```diff
@@ -65,6 +68,51 @@ impl WindowId {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct OsSpecificWindowEvent {
+    window: HWND,
+    msg: UINT,
+    wparam: WPARAM,
+    lparam: LPARAM,
+    retval: *mut Option<LRESULT>,
```

Comment:
Why `*mut` and not `&mut`? Sounds like a recipe for disaster, and `winit` tries to be as safe as it can be?

Reply:
@ZeGentzy  I know, but if we add a `'a` there, then we'll get `'a` everywhere (`WindowEvent`, `Event`, ...) and on every platform, i think?

Reply:
And currently this field is only accessed from `unsafe fn` members btw.

Reply:
Also, `&'a mut T` prevents cloning and copying.

Reply:
FWIW, we're probably going to add a lifetime to `WindowEvent` eventually, since there are some events that can only logically be handled through borrowing.

Reply:
Also, I realize now that this actually is a pretty big safety footgun. See, `send_event` doesn't always dispatch an event immediately: if `send_event` is called inside an active invocation of the user's event callback (which happens with certain `Window` functions), the event will get buffered and dispatched after the active callback invocation returns. If that happens when handling this type of event, `retval` will point to an invalid value when it's passed to the user's callback, and attempts to modify it will result in undefined behavior. The current API doesn't have any mechanisms to prevent that from happening.

Reply:
This will apply the same to the wparam and lparam arguments. If the handling is deferred, wparam and lparam (NMHDR or any pointers) will no longer be usable.

Is there anyway to prevent such message handling deferring?

Reply:
You could add a new variant of `send_event` that doesn't buffer events it can't dispatch. However, we can't get rid of deferred dispatch entirely, since it prevents the user callback from getting double-borrowed.